### PR TITLE
HomeWork 9

### PR DESCRIPTION
**Pull request by yurapukish**

### CODE REVIEW

File: Lessons9/HomeWork.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        # Step 1: Open the input.txt file and read it
+        with open('input.txt', 'r') as input_file:
+            text = input_file.read()
+
+            # Step 2: Find all matches using a regular expression
+            matches = re.findall(r'[a-zA-Z]', text)
+
+        # Step 3: Write the matches and their indices to the output file
+        output_file.writelines(f'{el:^7} -> {text.index(el):^7}\n'
```
Comment (antohaUa) [inline Lessons9/HomeWork.py]: Solution has gaps. You see if letters will be NOT unique - you will get incorrect positions. Just try to add additional 'e' at the beginning of input file. Thus probably findall not the best option here.

**Review by antohaUa — CHANGES_REQUESTED** [review]

**Comment by yurapukish** [comment]
hello, Update code
Change findall to finditer

File: Lessons9/HomeWork.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import re
+
+try:
+    # Create file to write the output
+    with open('output.txt', 'w') as output_file:
+        # Add a header to the output file
+        output_file.write(f'{"#" * 20}\n')
+
+        # Step 1: Open the input.txt file and read it
+        with open('input.txt', 'r') as input_file:
```
Comment (antohaUa) [inline Lessons9/HomeWork.py:25]: Please pay attention that Read/Write IO operations are time consuming. Thus better if possible reduce file operations and reduce time when file is open. In your case the best option read file first-> prepare data -> in separate block write to file. Not so critical , but just keep it in mind

File: Lessons9/HomeWork.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            text = input_file.read()
+
+            # Step 2: Find all matches using a regular expression
+            # Update here using finditer
+            matches = re.finditer(r'[a-zA-Z]', text)
+        # Step 3: Write the matches and their indices to the output file
+        output_file.writelines(f'{el.group():^7} -> {el.start():^7}\n'
```
Comment (antohaUa) [inline Lessons9/HomeWork.py:32]: For readability better split this line into two:   list comprehension first and then  output_file.writelines(lc)

File: Lessons9/HomeWork.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        # Add a header to the output file
+        output_file.write(f'{"#" * 20}\n')
+
+        # Step 1: Open the input.txt file and read it
+        with open('input.txt', 'r') as input_file:
+            text = input_file.read()
+
+            # Step 2: Find all matches using a regular expression
+            # Update here using finditer
+            matches = re.finditer(r'[a-zA-Z]', text)
```
Comment (antohaUa) [inline Lessons9/HomeWork.py:30]: exactly!
you see findall not able to cover all regex cases - so you always should choose between match/search/findall/finditer - to find smth which best suits your needs )

**Review by antohaUa — APPROVED** [review]

**Pull request merged**